std/random.d

```diff
@@ -1549,17 +1826,126 @@ how excellent the source of entropy is.
+                        {
+                            gamma = atomicLoad(_globalGamma);
+                            if (!gamma)
+                            {
+                                ulong[2] s = oneTimeInitSeedAndGamma();
+                                gamma = s[1];
+                                atomicStore(_globalSeed, s[0]);
+                                atomicStore(_globalGamma, gamma);
+                            }
+                        }
+                        assert(gamma);
+                    }
+                    // Stafford Mix05
+                    ulong x = atomicOp!"+="(_globalSeed, gamma);
+                    x = (x ^ (x >>> 31)) * 0x79c1_35c1_674b_9addUL;
+                    x = (x ^ (x >>> 29)) * 0x54c7_7c86_f691_3e45UL;
+                    return x ^ (x >>> 30);
+                }
+                else
+                {
+                    shared static uint _globalSeedLow;
+                    shared static uint _globalSeedHigh;
+                    shared static uint _globalGammaLow;
+                    shared static uint _globalGammaHigh;
+                    uint gammaLow = atomicLoad!(MemoryOrder.raw)(_globalGammaLow);
+                    if (!gammaLow)
+                    {
+                        synchronized
+                        {
+                            gammaLow = atomicLoad!(_globalGammaLow);
+                            if (!gammaLow)
+                            {
+                                ulong[2] s = oneTimeInitSeedAndGamma();
+                                atomicStore(_globalSeedHigh, cast(uint) (s[0] >>> 32));
+                                atomicStore(_globalSeedLow, cast(uint) s[0]);
+                                gammaLow = cast(uint) s[1];
+                                uint h = cast(uint) (s[1] >>> 32);
+                                assert(h);
+                                atomicStore(_globalGammaHigh, h);
+                                atomicStore(_globalGammaLow, gammaLow);
+                            }
+                        }
+                        assert(gammaLow);
+                    }
+                    uint gammaHigh = atomicLoad!(MemoryOrder.raw)(_globalGammaHigh);
+                    if (!gammaHigh)
+                    {
+                         gammaHigh = atomicLoad(_globalGammaHigh);
+                         assert(gammaHigh);
+                    }
+                    // Under contention updates can be interleaved. This is completely fine.
+                    uint low = atomicOp!"+="(_globalSeedLow, gammaLow);
+                    uint high = atomicOp!"+="(_globalSeedHigh, gammaHigh + uint(low < gammaLow));
+                    // Stafford Mix05
+                    ulong x = (cast(ulong) high) << 32 + cast(ulong) low;
+                    x = (x ^ (x >>> 31)) * 0x79c1_35c1_674b_9addUL;
+                    x = (x ^ (x >>> 29)) * 0x54c7_7c86_f691_3e45UL;
+                    return x ^ (x >>> 30);
```

Comment:
The `ulong` state => `ulong` variate transformation is repeated twice: it would nice to DRY this up in a single implementation called from both usage points.